data_diff/diff_tables.py

```diff
@@ -23,23 +23,36 @@ class TableSegment:
     database: Database
     table_path: DbPath
     key_column: str
-    extra_columns: Tuple[str, ...]
-    start: DbKey = None
-    end: DbKey = None
+    update_column: str = None
+    extra_columns: Tuple[str, ...] = ()
+    start_key: DbKey = None
+    end_key: DbKey = None
+    min_time: DbTime = None
+    max_time: DbTime = None
 
     _count: int = None
     _checksum: int = None
 
-    def _make_range_pred(self):
-        if self.start is not None:
-            yield Compare("<=", str(self.start), self.key_column)
-        if self.end is not None:
-            yield Compare("<", self.key_column, str(self.end))
+    def __post_init__(self):
+        if not self.update_column and (self.min_time or self.max_time):
+            raise ValueError("Error: min_time/max_time feature requires to specify 'update_column'")
+
+    def _make_key_range(self):
+        if self.start_key is not None:
+            yield Compare("<=", str(self.start_key), self.key_column)
+        if self.end_key is not None:
+            yield Compare("<", self.key_column, str(self.end_key))
+
+    def _make_update_range(self):
+        if self.min_time is not None:
+            yield Compare("<=", Time(self.min_time), self.update_column)
+        if self.max_time is not None:
+            yield Compare("<", self.update_column, Time(self.max_time))

```

Comment:
A Python question: Why is `yield` preferred here to just returning a `list`? Because it's annoying to merge the lists I'm assuming, versus adding generators together and then materializing them to a list like you're doing below? Kinda cool pattern.

Reply:
Yes, it's just a cleaner syntax, saving me from creating a list, appending to it, and returning it.